src/pkg/domains/files/files.go

```
@@ -68,23 +76,32 @@ func (d Domain) GetResources(ctx context.Context) (types.DomainResources, error)
 	// get a list of all the files we just downloaded in the temporary directory
 	files, err := listFiles(dst)
 	if err != nil {
```

if `listFiles()` returns an error should we return what we have?

---

I would return what we have - it's an accurate zero value.

caveat: I would also expect that an error here would expect would stop lula from running validate, but Lula currently swallows errors returned from GetResources if the domain was empty - I'm changing that behavior in a PR, but if you merge this right now you might see weird behavior .

proof/change: https://github.com/defenseunicorns/lula/pull/766/files#diff-2b1d27b989848257f397062966fca3725bac1057960904196ddcf28d17b0b73dR155

---

Appreciate the sanity check. 

Just to double back on your comment - we wouldn't exit a `validate` execution entirely due to one malformed `validation` yes?

Each validation should be a wholistic proof of evidence not coupled to other `validations` - Our stance on this is that there is still reason to complete the analysis of other controls/validations such that you do not require a full re-run in order to evaluate all areas of further evaluation.

---

Agreed, I think the goal is to return as many actual results as possible (I'd love to have a side conversation about "not-satisfied" results where something like "ERRORED" would be more informative, but that's neither here nor there)